### PR TITLE
[HotFix][Transform-v2][Spark] fix FieldMapper ClassCastException

### LINKED ISSUE #4081: [Bug][Transform-v2][Spark] FieldMapper ClassCastException
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.

### What happened

使用FakeSource生成数据执行输出到控制台没问题，使用FiledMap映射之后会报类型转换异常的错误

```
env {
  execution.parallelism = 1
  #spark.app.name = "SeaTunnel"
  #spark.executor.instances = 2
  #spark.executor.cores = 1
  #spark.executor.memory = "1g"
  #spark.master = local
}

source {
    FakeSource {
      result_table_name = "fake"
      row.num = 16
      schema = {
        fields {
          age = timestamp
          sex = date
        }
      }
    }
}

transform {
    FieldMapper {
        source_table_name = "fake"
        result_table_name = "result"
        field_mapper = {
            age = age
            sex = sex
        }
  }
}

sink {
  Console {
        source_table_name = "result"
  }
}
```

### SeaTunnel Version

dev分支

### SeaTunnel Config

```conf
seatunnel基本使用默认设置，使用seatunnel用spark3.4.1单节点去执行任务
```

### Running Command

```shell
../../bin/start-seatunnel-spark-3-connector-v2.sh  --master local[4]  --deploy-mode client  --config test
```

### Error Exception

```log
23/02/08 03:10:55 INFO CodeGenerator: Code generated in 29.388814 ms
23/02/08 03:10:55 ERROR SparkTaskExecuteCommand: Run SeaTunnel on spark failed.
org.apache.seatunnel.core.starter.exception.TaskExecuteException: SeaTunnel transform task: FieldMapper execute error
        at org.apache.seatunnel.core.starter.spark.execution.TransformExecuteProcessor.execute(TransformExecuteProcessor.java:93)
        at org.apache.seatunnel.core.starter.spark.execution.SparkExecution.execute(SparkExecution.java:61)
        at org.apache.seatunnel.core.starter.spark.command.SparkTaskExecuteCommand.execute(SparkTaskExecuteCommand.java:58)
        at org.apache.seatunnel.core.starter.SeaTunnel.run(SeaTunnel.java:39)
        at org.apache.seatunnel.core.starter.spark.SeaTunnelSpark.main(SeaTunnelSpark.java:31)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.apache.spark.deploy.JavaMainApplication.start(SparkApplication.scala:52)
        at org.apache.spark.deploy.SparkSubmit.org$apache$spark$deploy$SparkSubmit$$runMain(SparkSubmit.scala:958)
        at org.apache.spark.deploy.SparkSubmit.doRunMain$1(SparkSubmit.scala:180)
        at org.apache.spark.deploy.SparkSubmit.submit(SparkSubmit.scala:203)
        at org.apache.spark.deploy.SparkSubmit.doSubmit(SparkSubmit.scala:90)
        at org.apache.spark.deploy.SparkSubmit$$anon$2.doSubmit(SparkSubmit.scala:1046)
        at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:1055)
        at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)
Caused by: java.lang.ClassCastException: java.sql.Timestamp cannot be cast to java.lang.Long
        at org.apache.seatunnel.translation.spark.serialization.InternalRowConverter.reconvert(InternalRowConverter.java:194)
        at org.apache.seatunnel.translation.spark.serialization.InternalRowConverter.reconvert(InternalRowConverter.java:211)
        at org.apache.seatunnel.translation.spark.serialization.InternalRowConverter.reconvert(InternalRowConverter.java:187)
        at org.apache.seatunnel.translation.spark.serialization.InternalRowConverter.reconvert(InternalRowConverter.java:178)
        at org.apache.seatunnel.core.starter.spark.execution.TransformExecuteProcessor.sparkTransform(TransformExecuteProcessor.java:110)
        at org.apache.seatunnel.core.starter.spark.execution.TransformExecuteProcessor.execute(TransformExecuteProcessor.java:88)
        ... 16 more
![image](https://user-images.githubusercontent.com/46086591/217420595-e8f5b42c-0599-4a38-837d-6141f001357b.png)
```

### Flink or Spark Version

_No response_

### Java or Scala Version

_No response_

### Screenshots

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by FWLamb**
## Purpose of this pull request
close https://github.com/apache/incubator-seatunnel/issues/4081

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/incubator-seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by hailin0** [comment]
@FWLamb 
please add e2e testcase check this bug

#4081
https://github.com/apache/incubator-seatunnel/tree/dev/seatunnel-e2e/seatunnel-transforms-v2-e2e

**Comment by hailin0** [comment]
please resolve conflicts
@FWLamb

File: seatunnel-core/seatunnel-spark-starter/seatunnel-spark-2-starter/src/main/java/org/apache/seatunnel/core/starter/spark/execution/TransformExecuteProcessor.java [hunk 1]
```diff
@@ -125,6 +130,22 @@ private Dataset<Row> sparkTransform(SeaTunnelTransform transform, Dataset<Row> s
         transform.setTypeInfo(seaTunnelDataType);
         StructType structType =
                 (StructType) TypeConverterUtils.convert(transform.getProducedType());
+        StructField[] newStructFields =
```
Comment (Hisoka-X) [inline seatunnel-core/seatunnel-spark-starter/seatunnel-spark-2-starter/src/main/java/org/apache/seatunnel/core/starter/spark/execution/TransformExecuteProcessor.java:133]: Hi, can you move this logic to TypeConverterUtils.convert?

Comment (FWLamb) [reply seatunnel-core/seatunnel-spark-starter/seatunnel-spark-2-starter/src/main/java/org/apache/seatunnel/core/starter/spark/execution/TransformExecuteProcessor.java:133]: Modifications in TypeConverterUtils.convert may affect other modules.
The root cause of this problem is that for `TIMESTAMP` Type, the `TimestampType` in spark is returned in `TypeConverterUtils.convert`, while the `Long` Type is returned in `InternalRowConverter.convert`.
Inconsistent types cause spark to report an error.

**Comment by laglangyue** [comment]
please close this pr, it's fixed by pr #4335

**Pull request closed without merging**